api/unstable/fingerprint.h

```diff
@@ -0,0 +1,76 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <s2n.h>
+
+/**
+ * @file crl.h
+ *
+ * The following APIs enable applications to calculate fingerprints to
+ * identify ClientHellos.
+ *
+ * The fingerprinting APIs are currently considered unstable. They will be finalized
+ * and marked as stable after an initial customer integration and feedback.
+ */
+
+typedef enum {
+    /*
+     * The current standard open source fingerprinting method.
+     * See https://engineering.salesforce.com/tls-fingerprinting-with-ja3-and-ja3s-247362855967.
+     */
+    S2N_FINGERPRINT_JA3,
+} s2n_fingerprint_type;
+
+/**
+ * Calculates a fingerprint hash for a given ClientHello.
+ *
+ * Currently the only type supported is S2N_FINGERPRINT_JA3, which uses MD5 and
+ * requires at least 16 bytes of memory.
+ *
+ * @param ch The ClientHello to fingerprint.
+ * @param type The algorithm to use for the fingerprint. Currently only JA3 is supported.
+ * @param max_hash_size The maximum size of data that may be written to `hash`.
```

Comment:
I feel like this is kind of an obtuse way of saying that this parameter is the buffer size. Does this paragraph make it explicit to the user that they need to allocate some memory for the output hash?

Reply:
I mean, I'd say the function signature does? `hash` is a `uin8_t*`, not a `uin8_t**`, so it's not like the method could return allocated memory for the customer. And it feels weird to write out that the customer needs to allocate memory, especially since they could just use stack memory. Like, we don't know or care where you're getting `hash`.

I also think this follows the documentation pattern of our other APIs.